cluster-operator/src/main/java/io/strimzi/operator/cluster/Main.java

```diff
@@ -185,39 +186,48 @@ private static KafkaConnectS2IAssemblyOperator createS2iOperator(Vertx vertx, Ku
-            return Future.succeededFuture(config.isAssumeOpenShift());
-        } else if (client.isAdaptable(OkHttpClient.class)) {
+    static Future<PlatformFeaturesAvailability> getPlatformFeaturesAvailability(Vertx vertx, KubernetesClient client, ClusterOperatorConfig config)  {
+        if (client.isAdaptable(OkHttpClient.class)) {
             OkHttpClient ok = client.adapt(OkHttpClient.class);
-            Future<Boolean> fut = Future.future();
+            Future<PlatformFeaturesAvailability> fut = Future.future();
 
             vertx.executeBlocking(request -> {
-                try (Response resp = ok.newCall(new Request.Builder().get().url(client.getMasterUrl().toString() + "apis/route.openshift.io/v1").build()).execute()) {
-                    if (resp.code() >= 200 && resp.code() < 300) {
-                        log.debug("{} returned {}. We are on OpenShift.", resp.request().url(), resp.code());
-                        // We should be on OpenShift based on the /apis/route.openshift.io/v1 result. We can now safely try isAdaptable() to be 100% sure.
-                        Boolean isOpenShift = Boolean.TRUE.equals(client.isAdaptable(OpenShiftClient.class));
-                        request.complete(isOpenShift);
+                try {
+                    Boolean isOpenShift;
+                    if (config.isAssumeOpenShift() != null)  {
+                        log.debug("OpenShift has been set to {} through {}.", config.isAssumeOpenShift(), ClusterOperatorConfig.STRIMZI_ASSUME_OPENSHIFT);
+                        isOpenShift = config.isAssumeOpenShift();
                     } else {
-                        log.debug("{} returned {}. We are not on OpenShift.", resp.request().url(), resp.code());
-                        request.complete(Boolean.FALSE);
+                        Response resp = ok.newCall(new Request.Builder().get().url(client.getMasterUrl().toString() + "apis/route.openshift.io/v1").build()).execute();
+                        if (resp.code() >= 200 && resp.code() < 300) {
+                            log.debug("{} returned {}. We are on OpenShift.", resp.request().url(), resp.code());
+                            // We should be on OpenShift based on the /apis/route.openshift.io/v1 result. We can now safely try isAdaptable() to be 100% sure.
+                            isOpenShift = Boolean.TRUE.equals(client.isAdaptable(OpenShiftClient.class));
+                        } else {
+                            log.debug("{} returned {}. We are not on OpenShift.", resp.request().url(), resp.code());
+                            isOpenShift = false;
+                        }
                     }
+                    /* test */ String major = client.getVersion().getMajor().equals("") ? "1" : client.getVersion().getMajor();
+                    /* test */ String minor = client.getVersion().getMinor().equals("") ? "9" : client.getVersion().getMinor();
```

Comment:
What are these lines for?

Reply:
For some reason the ST at Travis `client.getVersion().getMajor()` and `client.getVersion().getMinor()` returned an empty strings.

Reply:
So why is it OK to assume that it's 1.9 _even when not running on Travis?_

Reply:
I was thinking when we cannot get info about version, we should not support any feature. I guess version like `0.0` would be better here. What do you think?

Reply:
I guess defaulting to the oldest supported might make sense.

Reply:
Alternatively we could support an env var which lets the user explicitly set the kube version to cover any case where the auto-detection does not work. Analogous to the assume openshift env var.

Reply:
That sounds like a good idea. I think it should be addressed in another PR, WDYT?